mypy/checker.py

```diff
@@ -5922,7 +5920,7 @@ def refine_away_none_in_comparison(
                 non_optional_types.append(typ)
 
         # Make sure we have a mixture of optional and non-optional types.
-        if len(non_optional_types) == 0 or len(non_optional_types) == len(chain_indices):
+        if not non_optional_types or len(non_optional_types) == len(chain_indices):
```

Comment:
This one I'd prefer to keep as is, as it retains symmetry with the condition on the right-hand side of the `or`